app/util/conf/base_config_loader.py

```diff
@@ -156,19 +164,23 @@ def _cast_and_set(self, key, value, config):
         :type config: Configuration
         """
         default_value = config.get(key)
-        if isinstance(default_value, int):
+
+        if isinstance(default_value, bool):
+            value_mapping = {'true': True, 'false': False}
+            if not isinstance(value, str) and not value.lower() in value_mapping.keys():
+                raise InvalidConfigError('The value for {} should be True or False, but it is "{}"'.format(key, value))
+            config.set(key, value_mapping[value.lower()])
+
+        elif isinstance(default_value, int):
```

Comment:
This is the actual fix -- just switched the order of the type checks so that `bool` is checked before `int`.